### ISSUE TITLE
crash at stsrtup

### ISSUE THREAD

**Issue by Smeagol001** [EmmittJ/PoESkillTree#449]
The error time: 3/4/2017 10:05 AM
Exception: System.NullReferenceException: Object reference not set to an instance of an object.
   at POESKillTree.Model.Builds.PoEBuild.DeepClone() in C:\GitHub\PoESkillTree_Source\WPFSkillTree\Model\Builds\PoEBuild.cs:line 281
   at POESKillTree.Model.Serialization.PersistentDataDeserializerCurrent.<DeserializeBuildAsync>d__14.MoveNext() in C:\GitHub\PoESkillTree_Source\WPFSkillTree\Model\Serialization\PersistentDataDeserializerCurrent.cs:line 238
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at POESKillTree.Model.Serialization.PersistentDataDeserializerCurrent.<DeserializeBuildsAsync>d__12.MoveNext() in C:\GitHub\PoESkillTree_Source\WPFSkillTree\Model\Serialization\PersistentDataDeserializerCurrent.cs:line 187
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at POESKillTree.Model.Serialization.PersistentDataDeserializerCurrent.<DeserializeFolderAsync>d__13.MoveNext() in C:\GitHub\PoESkillTree_Source\WPFSkillTree\Model\Serialization\PersistentDataDeserializerCurrent.cs:line 224
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at POESKillTree.Model.Serialization.PersistentDataDeserializerCurrent.<DeserializeBuildsAsync>d__11.MoveNext() in C:\GitHub\PoESkillTree_Source\WPFSkillTree\Model\Serialization\PersistentDataDeserializerCurrent.cs:line 169
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at POESKillTree.Model.Serialization.PersistentDataDeserializerCurrent.<DeserializeAdditionalFilesAsync>d__5.MoveNext() in C:\GitHub\PoESkillTree_Source\WPFSkillTree\Model\Serialization\PersistentDataDeserializerCurrent.cs:line 43
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at POESKillTree.Model.Serialization.AbstractPersistentDataDeserializer.<InitializeAsync>d__22.MoveNext() in C:\GitHub\PoESkillTree_Source\WPFSkillTree\Model\Serialization\AbstractPersistentDataDeserializer.cs:line 95
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at POESKillTree.Model.Serialization.PersistentDataSerializationService.PersistentData.<InitializeAsync>d__5.MoveNext() in C:\GitHub\PoESkillTree_Source\WPFSkillTree\Model\Serialization\PersistentDataSerializationService.cs:line 119
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at POESKillTree.Views.MainWindow.<Window_Loaded>d__82.MoveNext() in C:\GitHub\PoESkillTree_Source\WPFSkillTree\Views\MainWindow.xaml.cs:line 457
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.<>c.<ThrowAsync>b__6_0(Object state)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)

**Comment by brather1ng** [comment]
Could you please upload the build you had open when you last closed the program (or your whole build folder if you don't know which one that was)? The default build folder is `C:\Users\UserName\AppData\Roaming\PoESkillTree - Atlas of Worlds\Builds`.

**Comment by Smeagol001** [comment]
as i remember it was one of my whispering ice  builds
[Builds.zip](https://github.com/EmmittJ/PoESkillTree/files/818810/Builds.zip)

**Comment by brather1ng** [comment]
I couldn't reproduce the error you get (tried starting with each build). However, I got two other crashes, one
with a stack trace only differing from yours by one line, so that might have been the same problem.

I've fixed your build files so they don't crash my setup, changed Arc Templar and the Bladefall Totem builds: [Builds.zip](https://github.com/EmmittJ/PoESkillTree/files/818985/Builds.zip)

If this still crashes for you, I can't reproduce your issue.
